syncer.go

```diff
@@ -0,0 +1,41 @@
+package lookout
+
+import (
+	"context"
+	"fmt"
+
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/config"
+)
+
+// Syncer syncs the local copy of git repository for a given CommitRevision.
+type Syncer struct {
+	l *Library
+}
+
+// NewSyncer returns a Syncer for the given Library.
+func NewSyncer(l *Library) *Syncer {
+	return &Syncer{l}
+}
+
+// Sync syncs the local git repository to the given commit revision.
+func (s *Syncer) Sync(ctx context.Context, rev *CommitRevision) error {
+	r, err := s.l.GetOrInit(rev.Head.Repository)
+	if err != nil {
+		return err
+	}
+
+	opts := &git.FetchOptions{
```

Comment:
We don't need to execute a fetch if we already have the commit.

Reply:
But we need to create the branch, so...

Reply:
Only if it does not exist already.